streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamJoinWindow.java

```diff
@@ -27,14 +27,13 @@
 class KStreamJoinWindow<K, V> implements ProcessorSupplier<K, V> {
 
     private final String windowName;
-    private final long windowSizeMs;
-    private final long retentionPeriodMs;
-
 
     KStreamJoinWindow(String windowName, long windowSizeMs, long retentionPeriodMs) {
         this.windowName = windowName;
-        this.windowSizeMs = windowSizeMs;
-        this.retentionPeriodMs = retentionPeriodMs;
+
+        if (windowSizeMs * 2 > retentionPeriodMs)
+            throw new KafkaException("The retention period of the join window "
```

Comment:
If the exception in `AbstractStream` is a `TopologyBuilderException`, why isn't this one?